### PR TITLE
feat: add intialize and finalize for AstAnalyser.analyze API

### LINKED ISSUE #268: [AstAnalyser] Add initialize and finalize callback to manipulate SourceFile
Current AstAnalyser as flaw when we want to manipulate the SourceFile class before or after walking the AST, here is an example where we add new custom tracing;

```js
import { AstAnalyser } from "@nodesecure/js-x-ray";

const scanner = new AstAnalyser();

scanner.analyse("const foo = 'bar';", {
 initialize(sourceFile) {
   sourceFile.tracer.trace("...");
 },
 finalize(sourceFile) {
   // do something
 }
});
```

Those callbacks should be triggered before and after walking the AST

image

### PR DESCRIPTION
**Pull request by tony-go**
Fixes #268

Signed-off-by: Tony Gorez <gorez.tony@gmail.com>

### CODE REVIEW

**Comment by tony-go** [comment]
All tests should failed, I'd like to confirm that we are okay with the API first.

File: README.md [hunk 1]
```diff
@@ -223,6 +223,8 @@ interface RuntimeOptions {
   module?: boolean;
   removeHTMLComments?: boolean;
   isMinified?: boolean;
+  initialize?: (sourceFile: SourceFile) => void;
+  finalize?: (sourceFile: SourceFile) => void;
```
Comment (tony-go) [inline README.md:227]: More than documenting these types, I could add a proper example somewhere.

Comment (fraxken) [reply README.md:227]: probably somewhere in /docs/api/AstAnalyser.md

**Comment by fraxken** [comment]
Note: we need the same for analyseFile (it use analyse but interfaces/options should be the same)

**Comment by tony-go** [comment]
Hey @fraxken could you PTAL?

> Note: we need the same for analyseFile (it use analyse but interfaces/options should be the same)

Lets do it into a subsequent PR, if you don't mind.

**Pull request merged**